Simulator/Connections/Connections.cpp

```diff
@@ -24,11 +24,20 @@
 
 Connections::Connections()
 {
+   // Get a copy of the console logger to use in the case of errors
+   log4cplus::Logger consoleLogger_ = log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("console"));
+   
    // Create Edges/Synapses class using type definition in configuration file
    string type;
    ParameterManager::getInstance().getStringByXpath("//EdgesParams/@class", type);
    edges_ = Factory<AllEdges>::getInstance().createType(type);
 
+   // If the factory returns an error (nullptr), exit
+   if (!edges_) {
```

Comment:
This uses an implicit/automatic conversion of a pointer type to a `bool`. It's better, and clearer, to use `if (edges_ == nullptr)`